packages/jobs/lib/execution/operations/handler.ts

```diff
@@ -5,27 +5,47 @@ import { handleSyncError, handleSyncSuccess } from '../sync.js';
 import { handleWebhookError, handleWebhookSuccess } from '../webhook.js';
 import { toNangoError } from './utils/errors.js';
 
-import type { NangoProps, RunnerOutputError } from '@nangohq/types';
+import type { NangoProps, RunnerOutputError, TelemetryBag } from '@nangohq/types';
 import type { JsonValue } from 'type-fest';
 
-export async function handleSuccess({ taskId, nangoProps, output }: { taskId: string; nangoProps: NangoProps; output: JsonValue }): Promise<void> {
+export async function handleSuccess({
+    taskId,
+    nangoProps,
+    output,
+    telemetryBag
+}: {
+    taskId: string;
+    nangoProps: NangoProps;
+    output: JsonValue;
+    telemetryBag: TelemetryBag;
+}): Promise<void> {
     switch (nangoProps.scriptType) {
         case 'action':
-            await handleActionSuccess({ taskId, nangoProps, output });
+            await handleActionSuccess({ taskId, nangoProps, output, telemetryBag });
             break;
         case 'sync':
-            await handleSyncSuccess({ taskId, nangoProps });
+            await handleSyncSuccess({ taskId, nangoProps, telemetryBag });
             break;
         case 'webhook':
-            await handleWebhookSuccess({ taskId, nangoProps });
+            await handleWebhookSuccess({ taskId, nangoProps, telemetryBag });
             break;
         case 'on-event':
-            await handleOnEventSuccess({ taskId, nangoProps });
+            await handleOnEventSuccess({ taskId, nangoProps, telemetryBag });
```

Comment:
do we not have all the necessary data so we can publish the usage event here in one place?